### ISSUE TITLE
Animation on new messages

### ISSUE THREAD

**Issue by lmoureaux** [longturn/freeciv21#2038]
**Is your feature request related to a problem? Please describe.**

I often miss new messages.

**Describe the solution you'd like**

Ideally, I would like new messages to be displayed for a few seconds, just long enough to understand what they are about. They should then disappear.
Care should be taken not to overwhelm the user when messages are coming fast, such as during TC.

**Describe alternatives you've considered**

Blinking the messages button was mentioned.

**Additional context**

Get inspiration from messaging apps and desktop notifications?

**Comment by jwrober** [comment]
Where would this animation occur? On the main map in some kind of widget? We already have the icon that changes on the top menu bar. Maybe we have a "read" vs "unread" messages toggle of some kind so you can easily see which messages are new since you last opened the messages widget?

**Comment by lmoureaux** [comment]
This is to make sure that you see when you get new messages. The animation would be close to the messages widget or directly on top of it, so in the top-right corner presently.

**Comment by jwrober** [comment]
Kind of like the new turn fade-in/fade-out UI?

**Comment by lmoureaux** [comment]
Yes, but not as intrusive. Something small with the beginning of the message

**Comment by jwrober** [comment]
I'm not a fan of the blinking button after I did that for Diplomacy and the blow-back. Maybe make the icon change more obvious? 

Also, how do we handle mass-message events like bombardment or major battles?

**Comment by lmoureaux** [comment]
I'm not a fan of blink either. A single blink might work...

> Also, how do we handle mass-message events like bombardment or major battles?

Show at most 3-5 messages?

**Comment by jwrober** [comment]
I am not sure how to explain this is text, but here goes...

- Each message is truncated to max 128 chars or something small (first 128, not last)
- Each new message is shown below the messages button for a brief period (option set-able). Maybe default to a second or less.
- For more messages in the "view period" queue them below the previous message and as the view period expires for each they scroll up
- Don't show more than ? per cycle. That is another good option setting, but as many rapid fire messages come in they queue below the one before towards to bottom of the screen and as the old ones expire, the newer ones move up and eventually clear.

**Comment by jwrober** [comment]
Is this an RTS thing? What would happen if I came into an LT game and lost like 5 cities? Would I have to sit thought 50 seconds of queuing messages? I think this needs to be an RTS-like deal and not something that happens at login/connect during an LT game.

**Comment by lmoureaux** [comment]
> Each message is truncated to max 128 chars or something small (first 128, not last)

Even better, Qt has a way to elide text to a number of pixels.

> Maybe default to a second or less.

Sounds a bit short to quickly read text, but experience will tell.

> Don't show more than ? per cycle.

Maybe not show more than ? messages at a time and have a special animation to clear the screen when there are hidden messages.

> Is this an RTS thing?

Not only, players regularly complain about bugs explained in the messages.

> What would happen if I came into an LT game and lost like 5 cities? Would I have to sit thought 50 seconds of queuing messages? I think this needs to be an RTS-like deal and not something that happens at login/connect during an LT game.

Either you see 5 messages at most, or we add a special case to clear everything on login. (Or we could create a "what's new" welcome screen...)
